src/stores/external-source.ts

```diff
@@ -0,0 +1,138 @@
+import { derived, writable, type Readable, type Writable } from 'svelte/store';
+import type { UserId } from '../types/app';
+import {
+  type DerivationGroup,
+  type ExternalSourceSlim,
+  type ExternalSourceType,
+  type PlanDerivationGroup,
+} from '../types/external-source';
+import gql from '../utilities/gql';
+import { planId } from './plan';
+import { gqlSubscribable } from './subscribable';
+
+/* Writeable. */
+export const parsingError: Writable<string | null> = writable(null);
+export const creatingExternalSource: Writable<boolean> = writable(false);
+export const createExternalSourceError: Writable<string | null> = writable(null);
+export const createExternalSourceTypeError: Writable<string | null> = writable(null);
+export const createDerivationGroupError: Writable<string | null> = writable(null);
+export const derivationGroupPlanLinkError: Writable<string | null> = writable(null);
+export const getExternalSourceMetadataError: Writable<string | null> = writable(null);
+export const derivationGroupVisibilityMap: Writable<Record<DerivationGroup['name'], boolean>> = writable({});
+
+/* Subscriptions. */
+export const externalSources = gqlSubscribable<ExternalSourceSlim[]>(gql.SUB_EXTERNAL_SOURCES, {}, [], null);
+export const externalSourceTypes = gqlSubscribable<ExternalSourceType[]>(gql.SUB_EXTERNAL_SOURCE_TYPES, {}, [], null);
+export const derivationGroups = gqlSubscribable<DerivationGroup[]>(
+  gql.SUB_DERIVATION_GROUPS,
+  {},
+  [],
+  null,
+  transformDerivationGroups,
+);
+export const planDerivationGroupLinks = gqlSubscribable<PlanDerivationGroup[]>(
+  gql.SUB_PLAN_DERIVATION_GROUP,
+  {},
+  [],
+  null,
+);
+
+/* Derived. */
+// reorganization of planDerivationGroupLinks so that it is easy to grab entries by plan_id, and access the derivation groups and when their updates were last acknowledged
+export const derivationGroupsLastAcknowledged: Readable<
+  Record<number, { [derivation_group_name: string]: { acknowledged: boolean; last_acknowledged_at: string } }>
+> = derived(planDerivationGroupLinks, $planDerivationGroupLinks => {
+  const result: Record<
+    number,
+    { [derivation_group_name: string]: { acknowledged: boolean; last_acknowledged_at: string } }
+  > = {};
+  for (const entry of $planDerivationGroupLinks) {
```

Comment:
since it's already being loop through, can this only include the ones that haven't been acknowledged?

Reply:
Updated in [clean up filtering around $derivationGroupsAcknowledged](https://github.com/NASA-AMMOS/aerie-ui/commit/6dcd86bd787522eef2e99fedc376db1b0310cc0e).